cli/internal/state/state.go

```diff
@@ -20,32 +20,41 @@ import (
 
 	"dario.cat/mergo"
 	"github.com/edgelesssys/constellation/v2/internal/file"
+	"github.com/edgelesssys/constellation/v2/internal/validation"
 )
 
 const (
 	// Version1 is the first version of the state file.
 	Version1 = "v1"
 )
 
-// ReadFromFile reads the state file at the given path and returns the state.
+// ReadFromFile reads the state file at the given path and validates it.
+// If the state file is valid, the state is returned. Otherwise, an error
+// describing why the validation failed is returned.
 func ReadFromFile(fileHandler file.Handler, path string) (*State, error) {
 	state := &State{}
 	if err := fileHandler.ReadYAML(path, &state); err != nil {
 		return nil, fmt.Errorf("reading state file: %w", err)
 	}
+
+	v := validation.NewValidator()
+	if err := v.Validate(state, validation.ValidateOptions{}); err != nil {
+		return nil, fmt.Errorf("validating state file: %w", err)
+	}
```

Comment:
I think it would be beneficial to not validate the state file directly on read, but have separate functions/methods to do so, similar to how we currently trigger validation on the config file.
This would allow us to run different validation when opening the file for `constellation create` and for `constellation apply`.
The first requiring the file to be completely empty, the later requiring just the infrastructure to be set.
Other commands also rely on values from the cluster part of the file, so they might want to be able to validate that those values are set.

tldr; We should have separate validation functions that check for empty infrastructure or cluster values, as well as functions that do the opposite (check for valid infrastructure or valid cluster values) and call those from the CLI command functions.